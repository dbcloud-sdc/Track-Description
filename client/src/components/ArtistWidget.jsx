import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faUserFriends, faHeadphones, faCircle, faStar, faSquare, faExclamation,
} from '@fortawesome/free-solid-svg-icons';
import NumericLabel from 'react-pretty-numbers';
import FollowButton from './FollowButton';

import * as Styles from './artistStyles';

const params = {
  justification: 'L',
  locales: 'en-AU',
  currency: false,
  percentage: false,
  precision: 1,
  commafy: false,
  shortFormat: true,
  shortFormatMinValue: 1001,
  title: false,
};

function ArtistWidget({ artistData }) {
  if (!artistData) {
    return null;
  }

  function isPro(pro_status) {
    if (pro_status) {
      return (

        <span className="fa-layers fa-fw" style={{ marginLeft: '3px' }}>
          <span style={{ fontSize: '10px' }}>
            <FontAwesomeIcon icon={faCircle} size="m" color="#f50" />
          </span>
          <span style={{ fontSize: '8px' }}>
            <FontAwesomeIcon icon={faStar} size="xs" color="#fff" />
          </span>
        </span>
      );
    }
    return null;
  }

  const {
    avatar_picture, artist_name, no_of_followers, no_of_tracks, is_followed, pro_status,
  } = artistData;

  return (
    <Styles.artistWidget>
      <Styles.avatarPic src={avatar_picture} alt="avatar" />
      <Styles.artistName title={`Visit ${artist_name}'s Profile`}>
        {artist_name}
        {isPro(pro_status)}
      </Styles.artistName>
      <Styles.followAndTrackCount>
        <Styles.ArtistFollowBadge title={`${no_of_followers.toLocaleString()} followers`}>
          <FontAwesomeIcon color="#999" icon={faUserFriends} />
          {' '}
          <NumericLabel params={params}>{no_of_followers}</NumericLabel>
        </Styles.ArtistFollowBadge>
        <Styles.ArtistTrackBadge title={`${no_of_tracks} tracks`}>
          <FontAwesomeIcon color="#666" icon={faHeadphones} />
          {` ${no_of_tracks}`}
        </Styles.ArtistTrackBadge>

      </Styles.followAndTrackCount>
      <FollowButton isFollowing={is_followed} />
      <Styles.reportButton>
        <span className="fa-layers fa-fw">
          <FontAwesomeIcon icon={faSquare} size="1x" transform={{ rotate: 45 }} />
          <span style={{ fontSize: '7px' }}>
            <FontAwesomeIcon icon={faExclamation} size="1x" color="#fff" />
          </span>
        </span>
        {'  Report'}
      </Styles.reportButton>
    </Styles.artistWidget>

  );
}

export default ArtistWidget;
