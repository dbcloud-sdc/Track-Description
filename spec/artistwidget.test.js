
import React from 'react';
import Enzyme from 'enzyme';
import { mount, shallow, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ArtistWidget from '../client/src/components/ArtistWidget';

import artistData from '../client/src/data/artistData';

Enzyme.configure({ adapter: new Adapter() });

describe('ArtistWidget', () => {
  it('should render correctly in mode', () => {
    const component = shallow(<ArtistWidget artistData={artistData[0]} />);
    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
