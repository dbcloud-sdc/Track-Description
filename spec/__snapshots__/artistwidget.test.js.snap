// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistWidget should render correctly in mode 1`] = `
<div
  className="artistWidget"
>
  <img
    alt="avatar"
    src="https://s3.amazonaws.com/uifaces/faces/twitter/murrayswift/128.jpg"
  />
  <div
    id="artistName"
  >
    <span
      className="artistNameToolTipContainer"
      title="Visit ipsum's Profile"
    >
      ipsum
    </span>
  </div>
  <div
    className="followAndTrackCount"
  >
    <span
      className="ArtistFollowBadge"
      id={2134608}
      title="2,134,608 followers"
    >
      <span
        style={
          Object {
            "marginRight": "3px",
          }
        }
      >
        <FontAwesomeIcon
          border={false}
          className=""
          color="#999"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                640,
                512,
                Array [],
                "f500",
                "M192 256c61.9 0 112-50.1 112-112S253.9 32 192 32 80 82.1 80 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C51.6 288 0 339.6 0 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zM480 256c53 0 96-43 96-96s-43-96-96-96-96 43-96 96 43 96 96 96zm48 32h-3.8c-13.9 4.8-28.6 8-44.2 8s-30.3-3.2-44.2-8H432c-20.4 0-39.2 5.9-55.7 15.4 24.4 26.3 39.7 61.2 39.7 99.8v38.4c0 2.2-.5 4.3-.6 6.4H592c26.5 0 48-21.5 48-48 0-61.9-50.1-112-112-112z",
              ],
              "iconName": "user-friends",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </span>
      <NumericLabel
        params={
          Object {
            "commafy": false,
            "currency": false,
            "justification": "L",
            "locales": "en-AU",
            "percentage": false,
            "precision": 1,
            "shortFormat": true,
            "shortFormatMinValue": 1001,
            "title": false,
          }
        }
      >
        2134608
      </NumericLabel>
    </span>
    <span
      className="ArtistTrackBadge"
      title="23 tracks"
    >
      <span
        style={
          Object {
            "marginRight": "3px",
          }
        }
      >
        <FontAwesomeIcon
          border={false}
          className=""
          color="#666"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f025",
                "M256 32C114.52 32 0 146.496 0 288v48a32 32 0 0 0 17.689 28.622l14.383 7.191C34.083 431.903 83.421 480 144 480h24c13.255 0 24-10.745 24-24V280c0-13.255-10.745-24-24-24h-24c-31.342 0-59.671 12.879-80 33.627V288c0-105.869 86.131-192 192-192s192 86.131 192 192v1.627C427.671 268.879 399.342 256 368 256h-24c-13.255 0-24 10.745-24 24v176c0 13.255 10.745 24 24 24h24c60.579 0 109.917-48.098 111.928-108.187l14.382-7.191A32 32 0 0 0 512 336v-48c0-141.479-114.496-256-256-256z",
              ],
              "iconName": "headphones",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          padding-right="3px"
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </span>
      23
    </span>
  </div>
  <FollowButton
    isFollowing={0}
  />
  <div
    className="reportButton"
    style={
      Object {
        "marginTop": "30px",
      }
    }
  >
    <span
      className="fa-layers fa-fw"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f0c8",
              "M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z",
            ],
            "iconName": "square",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="1x"
        spin={false}
        symbol={false}
        title=""
        transform={
          Object {
            "rotate": 45,
          }
        }
      />
      <span
        style={
          Object {
            "fontSize": "7px",
          }
        }
      >
        <FontAwesomeIcon
          border={false}
          className=""
          color="#fff"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                192,
                512,
                Array [],
                "f12a",
                "M176 432c0 44.112-35.888 80-80 80s-80-35.888-80-80 35.888-80 80-80 80 35.888 80 80zM25.26 25.199l13.6 272C39.499 309.972 50.041 320 62.83 320h66.34c12.789 0 23.331-10.028 23.97-22.801l13.6-272C167.425 11.49 156.496 0 142.77 0H49.23C35.504 0 24.575 11.49 25.26 25.199z",
              ],
              "iconName": "exclamation",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="1x"
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </span>
    </span>
      Report
  </div>
</div>
`;
